packages/serve/src/createConfig.ts

```diff
@@ -18,11 +22,17 @@ export default function createConfig(args): devServerOptionsType {
         // clientLogging is not a valid devServer option
         delete options.clientLogging;
     }
-
-    if (options.hotOnly) {
-        options.hot = 'only';
-        // hotOnly is not a valid devServer option
-        delete options.hotOnly;
+    if (isDevServer4) {
+        if (options.hotOnly) {
+            options.hot = 'only';
+            // hotOnly is not a valid devServer option
+            delete options.hotOnly;
+        }
+    } else {
+        // only apply hotOnly when both are supplied
```

Comment:
```suggestion
        // when both hot flags are supplied, only hotOnly is applied
```

Reply:
`prefer --hotOnly over --hot`

Reply:
server seems to have logic for this so this is not needed anymore